ProcessLib/ComponentTransport/ComponentTransportProcess.cpp

```diff
@@ -73,14 +73,50 @@ void ComponentTransportProcess::assembleConcreteProcess(
     ProcessLib::ProcessVariable const& pv = getProcessVariables(process_id)[0];
 
     std::vector<std::reference_wrapper<NumLib::LocalToGlobalIndexMap>>
-        dof_table = {std::ref(*_local_to_global_index_map)};
+        dof_tables;
+    if (_use_monolithic_scheme)
+    {
+        dof_tables.push_back(std::ref(*_local_to_global_index_map));
+    }
+    else
+    {
+        setCoupledSolutionsOfPreviousTimeStep();
+        std::generate_n(
+            std::back_inserter(dof_tables), _process_variables.size(),
+            [&]() { return std::ref(*_local_to_global_index_map); });
+    }
     // Call global assembler for each local assembly item.
     GlobalExecutor::executeSelectedMemberDereferenced(
         _global_assembler, &VectorMatrixAssembler::assemble, _local_assemblers,
-        pv.getActiveElementIDs(), dof_table, t, x, M, K, b,
+        pv.getActiveElementIDs(), dof_tables, t, x, M, K, b,
         _coupled_solutions);
 }
 
+void ComponentTransportProcess::setCoupledSolutionsOfPreviousTimeStep()
+{
+    const auto number_of_coupled_solutions =
+        _coupled_solutions->coupled_xs.size();
+    _coupled_solutions->coupled_xs_t0.clear();
+    _coupled_solutions->coupled_xs_t0.reserve(number_of_coupled_solutions);
+    for (std::size_t i = 0; i < number_of_coupled_solutions; i++)
+    {
+        const auto& x_t0 = _xs_previous_timestep[i];
+        if (x_t0 == nullptr)
+        {
+            OGS_FATAL(
+                "Memory is not allocated for the global vector "
+                "of the solution of the previous time step for the ."
+                "staggered scheme.\n It can be done by overriding "
+                "Process::preTimestepConcreteProcess"
+                "(ref. "
+                "ComponentTransportProcess::preTimestepConcreteProcess) ");
```

Comment:
Is this message helpful for the user? What can the user do in such a situation?

Reply:
Worthless for users. I think we could discard the if-condition statement. What's your suggestion? @wenqing

Reply:
Removed.